src/CodeGen_Vulkan_Dev.cpp

```diff
@@ -0,0 +1,2919 @@
+            scalarize(op);
+
+        } else {
+            // Generate Phi node if used as an expression.
+            internal_assert(op->args.size() == 2 || op->args.size() == 3);
+            Expr else_expr;
+            if (op->args.size() == 3) {
+                else_expr = op->args[2];
+            }
+            SpvFactory::BlockVariables block_vars = emit_if_then_else(op->args[0], op->args[1], else_expr);
+            SpvId type_id = builder.declare_type(op->type);
+            SpvId result_id = builder.reserve_id(SpvResultId);
+            builder.append(SpvFactory::phi(type_id, result_id, block_vars));
+            builder.update_id(result_id);
+        }
+    } else if (op->is_intrinsic(Call::IntrinsicOp::div_round_to_zero)) {
+        internal_assert(op->args.size() == 2);
+        // See if we can rewrite it to something faster (e.g. a shift)
+        Expr e = lower_int_uint_div(op->args[0], op->args[1], /** round to zero */ true);
+        if (!e.as<Call>()) {
+            e.accept(this);
+            return;
+        }
+
+        SpvOp op_code = SpvOpNop;
+        if (op->type.is_float()) {
+            op_code = SpvOpFDiv;
+        } else if (op->type.is_int()) {
+            op_code = SpvOpSDiv;
+        } else if (op->type.is_uint()) {
+            op_code = SpvOpUDiv;
+        } else {
+            internal_error << "div_round_to_zero of unhandled type.\n";
+        }
+        visit_binary_op(op_code, op->type, op->args[0], op->args[1]);
+    } else if (op->is_intrinsic(Call::IntrinsicOp::mod_round_to_zero)) {
+        internal_assert(op->args.size() == 2);
+        SpvOp op_code = SpvOpNop;
+        if (op->type.is_float()) {
+            op_code = SpvOpFRem;  // NOTE: FRem matches the fmod we expect
+        } else if (op->type.is_int()) {
+            op_code = SpvOpSMod;
+        } else if (op->type.is_uint()) {
+            op_code = SpvOpUMod;
+        } else {
+            internal_error << "mod_round_to_zero of unhandled type.\n";
+        }
+        visit_binary_op(op_code, op->type, op->args[0], op->args[1]);
+
+    } else if (op->is_intrinsic(Call::shift_right)) {
+        internal_assert(op->args.size() == 2);
+        if (op->type.is_uint() || (op->args[1].type().is_uint())) {
+            visit_binary_op(SpvOpShiftRightLogical, op->type, op->args[0], op->args[1]);
+        } else {
+            Expr e = lower_signed_shift_right(op->args[0], op->args[1]);
+            e.accept(this);
+        }
+    } else if (op->is_intrinsic(Call::shift_left)) {
+        internal_assert(op->args.size() == 2);
+        if (op->type.is_uint() || (op->args[1].type().is_uint())) {
+            visit_binary_op(SpvOpShiftLeftLogical, op->type, op->args[0], op->args[1]);
+        } else {
+            Expr e = lower_signed_shift_left(op->args[0], op->args[1]);
+            e.accept(this);
+        }
+    } else if (op->is_intrinsic(Call::strict_float)) {
+        // TODO: Enable/Disable RelaxedPrecision flags?
```

Comment:
Do we need a github issue for this todo?

Reply:
Issue added ... tracking this under:

[vulkan] Add support for strict_float / relaxed_precision in SPIR-V #7239